### PR TITLE
[test] Fix flaky test

### PR DESCRIPTION
**Pull request by oliviertassinari**
The flakiness rate seems to be above 10%. It's too much for a pleasant DX. It should be fixed. The fix is to make `useData` systematically resolve when wrapped with `act()`: https://reactjs.org/docs/test-utils.html#act, it requires to remove the usage of the promise.

I have also worked on it to signal that skipping is almost always the wrong answer :D.

### CODE REVIEW

File: packages/grid/x-grid-modules/src/components/cell.tsx [hunk 1]
```diff
@@ -51,7 +51,6 @@ export const Cell: React.FC<GridCellProps> = React.memo((props) => {
       role="cell"
       data-value={value}
       data-field={field}
-      data-colindex={colIndex}
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/components/cell.tsx:54]: Duplicates with `aria-colindex` better use the a11y as the source of truth for the logic and prevent anyone to rely on arbitrary data attributes.

File: packages/grid/x-grid-modules/src/hooks/root/useEvents.ts [hunk 2]
```diff
@@ -72,8 +71,8 @@ export function useEvents(
         const id = getIdFromRowElem(rowEl);
         const rowModel = apiRef.current.getRowFromId(id);
         const rowIndex = apiRef.current.getRowIndexFromId(id);
-        const field = getDataFromElem(cellEl, 'field');
-        const value = getDataFromElem(cellEl, 'value');
+        const field = cellEl.getAttribute('data-field');
+        const value = cellEl.getAttribute('data-value');
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/hooks/root/useEvents.ts:75]: IMHO, an unnecessary abstraction solves no bug, makes it harder to understand what's going on.

File: packages/grid/x-grid/src/XGrid.test.tsx [hunk 3]
```diff
@@ -114,9 +120,8 @@ describe('<XGrid />', () => {
 
     it('cell navigation with arrows ', async () => {
       render(<KeyboardTest />);
-      await raf();
       // @ts-ignore
-      document.querySelector('[data-rowindex="0"]').querySelector('[data-colindex="0"]').focus();
+      document.querySelector('[role="cell"][data-rowindex="0"][aria-colindex="0"]').focus();
```
Comment (oliviertassinari) [inline packages/grid/x-grid/src/XGrid.test.tsx:124]: Reduce the potential for confusion. We query a cell.

File: packages/grid/x-grid/src/XGrid.test.tsx [hunk 4]
```diff
@@ -138,9 +143,8 @@ describe('<XGrid />', () => {
 
     it('Home / End navigation', async () => {
       render(<KeyboardTest />);
-      await raf();
```
Comment (oliviertassinari) [inline packages/grid/x-grid/src/XGrid.test.tsx:141]: This await was unreliable, it was leading to the flaky test.

**Review by dtassone — APPROVED** [review]

**Pull request merged**